libs/dal/src/repositories/base-repository.ts

```diff
@@ -1,11 +1,11 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { ClassConstructor, plainToInstance } from 'class-transformer';
-import { Document, Model, Query, Types, ProjectionType, FilterQuery } from 'mongoose';
+import { Model, Types, ProjectionType, FilterQuery, UpdateQuery } from 'mongoose';
 
-export class BaseRepository<T_Query, T_Response> {
-  public _model: Model<any & Document>;
+export class BaseRepository<T_DBModel, T_MappedEntity> {
```

Comment:
Did a lot of type fixes here (no logic changes), but basically, the `BaseRepository` requires now to pass the `T_DBModel` which is a DB model, and `T_MappedEntity` which is the type that we map the DB object to. 
So like the Mongoose model return `DBModel` objects and we map them to `Entity`, just to have ids as strings, not the ObjectIds.